accord-core/src/main/java/accord/local/CommonAttributes.java

```diff
@@ -84,6 +80,7 @@ public TxnId txnId()
 
         public Mutable txnId(TxnId txnId)
         {
+            Invariants.checkState(this != EMPTY_ATTRS && this != EMPTY_ATTRS_NONE_PARTIAL_DEPS, "Don't mutate the global empty instances");
```

Comment:
in this case I kinda feel extending `CommonAttributes.Mutable` might be best... though its more code


```
private static class ImmutableAttributes extends CommonAttributes.Mutable
{
    priviate final String name; // to know if this was empty or empty w/ deps.none
```

Reply:
I'll remove the EMPTY instances so it won't matter